### PR TITLE
ptl-tool: load labeled PRs

### PR DESCRIPTION
**Pull request by batrick**
This lets you label in GitHub the PRs you want to test/merge.

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

### CODE REVIEW

File: src/script/ptl-tool.py [hunk 1]
```diff
@@ -43,13 +43,24 @@
...
+#
+#
+# Fetching PRs by label:
+#
+# $ src/script/ptl-tool.py --base master --branch HEAD --merge-branch master --label - --pr-label wip-pdonnell-testing
+# Detaching HEAD onto base: master
+# Adding labeled PR #18192 to PR list
+# Will merge PRs: [18192]
+# Merging PR #18192
+# Leaving HEAD detached; no branch anchors your commit
```
Comment (joscollin) [inline src/script/ptl-tool.py]: I think we can add this simple example  too:

`src/script/ptl-tool.py --base master --merge-branch <wip-test-branch> --pr-label wip-<name>-testing`

**Comment by joscollin** [comment]
The initial test looks fine. But I will do more testing :-)

File: src/script/ptl-tool.py [hunk 2]
```diff
@@ -141,9 +152,24 @@ def build_branch(args):
         # So we know that we're not on an old test branch, detach HEAD onto ref:
         base.checkout()
 
-    for pr in args.prs:
-        log.info("Merging PR #{pr}".format(pr=pr))
+    prs = args.prs
+    if args.pr_label:
+        payload = {'labels': args.pr_label, 'sort': 'created', 'direction': 'desc'}
+        labeled_prs = requests.get("https://api.github.com/repos/ceph/ceph/issues", auth=(USER, PASSWORD), params=payload)
```
Comment (joscollin) [inline src/script/ptl-tool.py]: See, created the test branch for an invalid pr-label.
```
$ src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label wip-jcollin-testing2 #or wip-aoeu-testing
Detaching HEAD onto base: master
Will merge PRs: []
Checked out new branch wip-jcollin-testing-20171011
Created tag testing/wip-jcollin-testing-20171011_04
jcollin@smithi042:~/test-ptl-oct11$ git branch
  master
  * wip-jcollin-testing-20171011
```

Please check if the branch exists first and drop it. May be move the existing label validation code to a function ?

File: src/script/ptl-tool.py (hunk 1 above)
Comment (batrick) [reply src/script/ptl-tool.py]: I'm trying not to have too many examples. :)

File: src/script/ptl-tool.py (hunk 2 above)
Comment (batrick) [reply src/script/ptl-tool.py]: Fixed!

Comment (joscollin) [reply src/script/ptl-tool.py]: `--pr-label ' '`is bypassing it. I have pressed Ctrl+C and it saved me :-D. 

```
jcollin@smithi042:~/test-ptl-oct12$ src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label ' '
Adding labeled PR #18271 to PR list
Adding labeled PR #18270 to PR list
Adding labeled PR #18269 to PR list
Adding labeled PR #18268 to PR list
Adding labeled PR #18267 to PR list
Adding labeled PR #18266 to PR list
Adding labeled PR #18265 to PR list
Adding labeled PR #18264 to PR list
Adding labeled PR #18263 to PR list
Adding labeled PR #18262 to PR list
Adding labeled PR #18261 to PR list
Adding labeled PR #18260 to PR list
Adding labeled PR #18259 to PR list
Adding labeled PR #18257 to PR list
Adding labeled PR #18256 to PR list
Adding labeled PR #18255 to PR list
Adding labeled PR #18254 to PR list
Adding labeled PR #18253 to PR list
Adding labeled PR #18251 to PR list
Adding labeled PR #18250 to PR list
Adding labeled PR #18249 to PR list
Adding labeled PR #18248 to PR list
Adding labeled PR #18247 to PR list
Adding labeled PR #18246 to PR list
Adding labeled PR #18245 to PR list
Adding labeled PR #18244 to PR list
Adding labeled PR #18243 to PR list
Adding labeled PR #18241 to PR list
Adding labeled PR #18240 to PR list
Adding labeled PR #18239 to PR list
Will merge PRs: [18271, 18270, 18269, 18268, 18267, 18266, 18265, 18264, 18263, 18262, 18261, 18260, 18259, 18257, 18256, 18255, 18254, 18253, 18251, 18250, 18249, 18248, 18247, 18246, 18245, 18244, 18243, 18241, 18240, 18239]
Detaching HEAD onto base: master
Merging PR #18271
Labeled PR #18271 wip-jcollin-testing
Merging PR #18270
^CTraceback (most recent call last):
  File "src/script/ptl-tool.py", line 308, in <module>
    
  File "src/script/ptl-tool.py", line 305, in main

```

Comment (batrick) [reply src/script/ptl-tool.py]: okay, fixed

Comment (joscollin) [reply src/script/ptl-tool.py]: @batrick Thanks. I will test tomorrow. Need some sleep now. :-)

Comment (joscollin) [reply src/script/ptl-tool.py]: @batrick In the mean time, could you clarify what is `label = '-'` ? It is also an invalid label, right ?

Comment (joscollin) [reply src/script/ptl-tool.py]: 1. Check the empty string also.
```
$ src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label ''
Will merge PRs: []
Detaching HEAD onto base: master
Checked out new branch wip-jcollin-testing-20171013
Created tag testing/wip-jcollin-testing-20171013
```

> Adding labeled PR #18284 to PR list

2. Do we need to add and merge an unreviewed / unapproved PR for testing ? Do we need to check the PR status before the add / merge ?

Comment (batrick) [reply src/script/ptl-tool.py]: > Check the empty string also.
done.

> Do we need to add and merge an unreviewed / unapproved PR for testing ? Do we need to check the PR status before the add / merge ?

No, that should stay the ptl's responsibility to check.

**Comment by joscollin** [comment]
@batrick I did this to merge `wip-ptltool1-testing` PRs to my testing branch:
`$src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label wip-ptltool1-testing`

Then I want to merge `wip-ptltool2-testing` also to the same testing branch. So I did:
```
$ src/script/ptl-tool.py --base-path refs/heads/ --base wip-jcollin-testing-20171016 --branch wip-jcollin-testing-20171016 wip-ptltool2-testing
usage: ptl-tool.py [-h] [--branch BRANCH]
                   [--merge-branch-name MERGE_BRANCH_NAME] [--base BASE]
                   [--base-path BASE_PATH] [--git-dir GIT] [--label LABEL]
                   [--pr-label PR_LABEL]
                   [PR [PR ...]]
ptl-tool.py: error: argument PR: invalid int value: 'wip-ptltool2-testing'
```

Is this expected ? This is working as expected, if I try with the PR number:
`$ src/script/ptl-tool.py --base-path refs/heads/ --base wip-jcollin-testing-20171016 --branch wip-jcollin-testing-20171016 18107`

But not with the label feature.

**Comment by batrick** [comment]
> @batrick I did this to merge `wip-ptltool1-testing` PRs to my testing branch:
? `$src/script/ptl-tool.py --base master --merge-branch wip-jcollin-testing --pr-label wip-ptltool1-testing`

There is no `--merge-branch` option.

> Then I want to merge `wip-ptltool2-testing` also to the same testing branch. So I did:
```
$ src/script/ptl-tool.py --base-path refs/heads/ --base wip-jcollin-testing-20171016 --branch wip-jcollin-testing-20171016 wip-ptltool2-testing
usage: ptl-tool.py [-h] [--branch BRANCH]
                   [--merge-branch-name MERGE_BRANCH_NAME] [--base BASE]
                   [--base-path BASE_PATH] [--git-dir GIT] [--label LABEL]
                   [--pr-label PR_LABEL]
                   [PR [PR ...]]
ptl-tool.py: error: argument PR: invalid int value: 'wip-ptltool2-testing'
```

Looks like you forgot `--pr-label wip-ptltool2-testing`?

**Comment by joscollin** [comment]
@batrick Yes, I missed `--pr-label`. Sorry for the negligence, I didn't even check the error message.

**Comment by joscollin** [comment]
@batrick Oh, I think that is intended. No testing branch.

**Comment by joscollin** [comment]
Merged as this is an independent script (doesn't break anything else).

**Pull request merged**